packages/core-api/lib/versions/2/schema/blocks.js

```diff
@@ -11,10 +11,10 @@ exports.index = {
     ...pagination,
     ...{
       orderBy: Joi.string(),
-      id: Joi.string(),
+      id: Joi.string().regex(/^[0-9]+$/, 'numbers'),
```

Comment:
Block IDs can probably be validated as `Joi.number().integer().unsafe()`.

Reply:
That should probably suffice, as there are no side effects with invalid inputs anyway.